src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```diff
@@ -376,13 +379,16 @@ private ProjectRepo getUpdatedRepoCritical(
             String migratedFromID = doc.getMigratedFromID();
             if (migratedFromID != null) {
                 Log.info("[{}] Has a migratedFromId: {}", projectName, migratedFromID);
-                try (LockGuard __ = lock.lockGuard(migratedFromID)) {
+                ProjectContext context = ContextStore.Instance().getContextForProject(migratedFromID);
```

Comment:
To check my understanding... will the inner 'migrated from' project get a different database connection compared to the outer 'migrating to' project, because they have separate project contexts?

Reply:
Yes, each project will have it's own connection/lock :+1: